src/Phan/Analysis/PostOrderAnalysisVisitor.php

```diff
@@ -588,20 +589,21 @@ public function visitGlobal(Node $node): Context
         $optional_global_variable_type = Variable::getUnionTypeOfHardcodedGlobalVariableWithName($variable_name);
         if ($optional_global_variable_type) {
             $variable->setUnionType($optional_global_variable_type);
+            $scope_global_variable = $variable;
         } else {
             $scope = $this->context->getScope();
-            if ($scope->hasGlobalVariableWithName($variable_name)) {
-                // TODO: Support @global, add a clone to the method context?
-                $actual_global_variable = clone($scope->getGlobalVariableByName($variable_name));
-                $actual_global_variable->setUnionType($actual_global_variable->getUnionType()->eraseRealTypeSetRecursively());
-                $this->context->addScopeVariable($actual_global_variable);
-                return $this->context;
+            if (!$scope->hasGlobalVariableWithName($variable_name)) {
+                $this->context->addGlobalScopeVariable( clone $variable );
             }
+            // TODO: Support @global?
+            $actual_global_variable = $scope->getGlobalVariableByName($variable_name);
+            $scope_global_variable = new GlobalVariable($actual_global_variable);
```

Comment:
Top-level file scopes are different from the global scope, although Phan pretends they're the same thing.

E.g. you can use `global $var` in a file scope.

```php
<?php
global $var;
```

```suggestion
            $scope_global_variable = $actual_global_variable instanceof GlobalVariable ? clone($actual_global_variable) : new GlobalVariable($actual_global_variable);
```